hindsight-api/hindsight_api/engine/llm_wrapper.py

```diff
@@ -224,11 +271,31 @@ async def call(
                                     schema_msg + "\n\n" + call_params["messages"][0]["content"]
                                 )
 
-                        call_params["response_format"] = {"type": "json_object"}
+                        # LM Studio and Ollama don't support json_object response format reliably
+                        # We rely on the schema in the system message instead
+                        if self.provider not in ("lmstudio", "ollama"):
+                            call_params["response_format"] = {"type": "json_object"}
+                        
+                        logger.info(f"[LLM DEBUG] Sending request to {self.provider}/{self.model} (timeout={getattr(self, 'timeout', 'None')})")
                         response = await self._client.chat.completions.create(**call_params)
+                        logger.info(f"[LLM DEBUG] Received response from {self.provider}/{self.model}")
 
                         content = response.choices[0].message.content
-                        json_data = json.loads(content)
+                        
+                        # For local models, they may wrap JSON in markdown code blocks
+                        if self.provider in ("lmstudio", "ollama"):
```

Comment:
this is a good thing, can we do it for all the providers? 
we don't enforce the schema checks because some models are too stupid and couldn't get it right all the time.